x/protorev/keeper/epoch_hook_test.go

```diff
@@ -117,6 +120,55 @@ func (suite *KeeperTestSuite) TestEpochHook() {
 	suite.Require().Equal(totalNumberExpected, totalActuallySeen)
 }
 
+func (suite *KeeperTestSuite) TestGetHighestLiquidityPool() {
+	testCases := []struct {
+		name                   string
+		inputBaseDenomPools    map[string]map[string]keeper.LiquidityPoolStruct
+		expectedBaseDenomPools map[string]map[string]keeper.LiquidityPoolStruct
+	}{
+		{
+			// There are 2 GAMM pools with epochOne as a denom, with poolId 47 being the one with the higher liquidity
+			// So we force epochOne to be considered a base denom as input, to assess the method chooses the correct pool
+			// Within the same pool module
+			name: "Get highest liquidity pools for two GAMM pools",
+			inputBaseDenomPools: map[string]map[string]keeper.LiquidityPoolStruct{
+				"epochOne": {},
+			},
+			expectedBaseDenomPools: map[string]map[string]keeper.LiquidityPoolStruct{
+				"epochOne": {
+					"uosmo": {Liquidity: sdk.NewInt(2000000), PoolId: 47},
+				},
+			},
+		},
+		{
+			// There is 1 GAMM pool and 1 Concentrated Liquidity pool with epochTwo as a denom
+			// With pool id 49 being the one with the higher liquidity
+			// So we force epochTwo to be considered a base denom as input, to assess the method chooses the correct pool
+			// Across different pool modules
+			name: "Get highest liquidity pools for one GAMM pool and one Concentrated Liquidity pool",
+			inputBaseDenomPools: map[string]map[string]keeper.LiquidityPoolStruct{
+				"epochTwo": {},
+			},
+			expectedBaseDenomPools: map[string]map[string]keeper.LiquidityPoolStruct{
+				"epochTwo": {
+					"uosmo": {Liquidity: sdk.NewInt(2000000), PoolId: 49},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			err := suite.App.ProtoRevKeeper.GetHighestLiquidityPools(suite.Ctx, tc.inputBaseDenomPools)
+			suite.Require().NoError(err)
+			suite.Require().Equal(tc.inputBaseDenomPools, tc.expectedBaseDenomPools)
```

Comment:
We need to be more explicit / deliberate with these tests. By just looking at this code, the reviewer has no way to know that SetupTest creates a ton of pools where GetHighestLiquidityPools can then fetch from. We should also say in the description of the test case which pool IDs are created that consist of the base denom and what their respective liquidity values are

Reply:
@czarcas7ic Do you want us to create the pools within this test instead of in SetupTest(), or do you want a comment that explains that SetupTests creates pools?

The way the test suite for protorev is set up right now is that SetupTest creates all the pools since many tests require specifying pool ids, so for us to keep a single order we use SetupTest to create all pools for all tests. 

To the second point of adding a comment in the description, agreed and can do so.

Reply:
A comment is fine! My point is, everywhere else in the code, SetupTest does not create pools. So when I saw this test, I didn't even think to look there. I then made some print lines, and realized what was happening, so looked in SetupTest and finally figured out what was going on. I would like to save any one else reviewing in the future those five minutes is all 😅

Reply:
So basically just asking for a commed above SetupTest explaining what is going on in relation to pool creation for the getter

Reply:
Mmm, also, this makes me think... Should this really be called a "getter"? Its modifying one of the inputs right?

Reply:
This all makes sense to me, sorry you spent that time and it wasn't documented better when first worked on!

Improved comments here: https://github.com/osmosis-labs/osmosis/pull/4946/commits/957b64f95f1de58954be42cb053f30f2767ab552

Changed name from Get to Update here: 
https://github.com/osmosis-labs/osmosis/pull/4946/commits/c9ab2084f1f5851bf9859856da418bef0cdb0a3a

Reply:
No problem at all! Thanks for the swift change 🙂